### PR TITLE
Setup sbt-ci-release

### PR DESCRIPTION
**Pull request by cquiroz**
This should allow us to release based on `sbt-ci-relese` and directly from github actions.
We'd need to setup the gpg keys and sonatype access credentials though, see

https://github.com/olafurpg/sbt-ci-release

I don't have enough right to put secrets though.
This PR may need several iterations to get it right

### CODE REVIEW

**Comment by cquiroz** [comment]
@sjrd @julienrf I'd need your help to set this up, basically by setting the variables on GitHub

**Comment by sjrd** [comment]
I'll have a look next week. I'm afraid I'm completely busy this week.

**Comment by ngbinh** [comment]
@sjrd can we move this forward?

**Comment by sjrd** [comment]
I am looking at this today, yes.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          path: ~/.sbt
+          key: ${{ runner.os }}-sbt-${{ hashFiles('build.sbt') }}-${{ hashFiles('project/*.scala') }}
+          restore-keys: ${{ runner.os }}-sbt-
+      - name: Publish
+        run: sbt clean ci-release
+        env:
+          PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
```
Comment (sjrd) [inline .github/workflows/release.yml:31]: You'll have to also add:
```
CI_RELEASE: "; scalajs-bundler-linker/publishSigned; sbt-scalajs-bundler/publishSigned; sbt-web-scalajs-bundler/publishSigned"
CI_SNAPSHOT_RELEASE: "; scalajs-bundler-linker/publish; sbt-scalajs-bundler/publish; sbt-web-scalajs-bundler/publish"
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          path: ~/.cache/coursier/v1
+          key: ${{ runner.os }}-coursier-${{ hashFiles('build.sbt') }}-${{ hashFiles('project/*.scala') }}
+          restore-keys: ${{ runner.os }}-coursier-
+      - name: Cache .sbt
+        uses: actions/cache@v1
+        with:
+          path: ~/.sbt
+          key: ${{ runner.os }}-sbt-${{ hashFiles('build.sbt') }}-${{ hashFiles('project/*.scala') }}
+          restore-keys: ${{ runner.os }}-sbt-
```
Comment (sjrd) [inline .github/workflows/release.yml:24]: Is there a reason this is different than what we have in our existing https://github.com/scalacenter/scalajs-bundler/blob/master/.github/workflows/ci.yml ?

**Review by sjrd — CHANGES_REQUESTED** [review]
I have set up the appropriate Repository Secrets now. I have a couple comments to this PR.

**Comment by cquiroz** [comment]
Updated according to the comments

**Comment by sjrd** [comment]
Thanks. Could you squash everything in a single commit with a clean commit message?

**Comment by cquiroz** [comment]
ok, squashed

**Review by sjrd — APPROVED** [review]

**Pull request merged**